uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/absence-tab/components/absence-tab-work-patterns.js

```diff
@@ -58,12 +65,17 @@ define([
       });
     };
 
-    // @TODO -This is temporary to open the modal, test cases are pending
-    // Will be fixed in PCHR-2016
+    /**
+     * Opens the Custom Work Pattern Modal
+     */
     vm.openModal = function () {
       $uibModal.open({
         appendTo: $rootElement.children().eq(0),
-        templateUrl: settings.pathTpl + 'components/absence-tab-custom-work-pattern-modal.html'
+        template: '<absence-tab-custom-work-pattern-modal dismiss="$ctrl.dismiss()" contact-id="' + vm.contactId + '"/>',
+        controller: [ '$uibModalInstance', function ($modalInstance) {
```

Comment:
if you're giving a self-contained component to `$uibModal` why is it needed to define also an additional controller?